src/OrbitGl/App.cpp

```diff
@@ -1017,17 +1018,23 @@ void OrbitApp::LoadModuleOnRemote(ModuleData* module_data,
               absl::StrFormat("Did not find symbols locally or on remote for module \"%s\": %s\n%s",
                               module_data->file_path(), error_message_from_local,
                               result.error().message()));
-          main_thread_executor_->Schedule([this, module_data]() {
-            modules_currently_loading_.erase(module_data->file_path());
-          });
+
+          auto executor = main_thread_executor.lock();
+          if (executor != nullptr) {
```

Comment:
Shouldn't you also invert that check and do an early return, as below?

Reply:
I debated that with myself. Because If I return early I have the `return` twice and only a single statement in between. What do you prefer?

Reply:
You are probably going to replace this with TrySchedule anyways?

Reply:
> I debated that with myself. Because If I return early I have the `return` twice and only a single statement in between. What do you prefer?

I have a slight preference for the early return. But I guess both have valid pros and cons.

Reply:
BTW. I agreed with Dimitry on the other branch to introduce a function `TrySchedule` which takes a weak_ptr to an Executor and a function object. With this function, the whole lock-and-check procedure will go away. I want to rebase this PR, when the other one is merged. So this discussion might eventually be obsolete.

Reply:
I will replace it when the other branch is merged.